apps/publisher/src/App.tsx

```diff
@@ -35,6 +39,26 @@ function App() {
 
   const { startStreaming, stopStreaming, publisherState } = usePublisher(accessToken, streamId);
 
+  const {cameraList, microphoneList, camera, microphone, setCamera, setMicrophone, mediaStream} = useMediaDevices();
+
+  const video = useRef<HTMLVideoElement>(null)
+
+  useEffect(() => {
+    if (video.current && (mediaStream instanceof MediaStream)) {
```

Comment:
only need to check if it's undefined here, because it's declared in useMediaDevices hook as MediaStream | undefined